beacon-chain/rpc/beacon/validators.go

```diff
@@ -209,8 +209,20 @@ func (bs *Server) ListValidators(
 		)
 	}
 
-	validatorCount := len(validators)
-	// If there are no validators, we simply return a response specifying this.
+	// Filter active validators if the request specifies it.
+	res := vals
+	if req.Active {
+		filteredValidators := make([]*ethpb.Validator, 0)
+		for _, val := range vals {
```

Comment:
we can filter in the above loop

Reply:
Hey @shayzluf I tried to do it in the above loop but that's only for archival data, so I would need another loop for non-archival requests anyways, creating more code than necessary. This seemed to be the shortest approach.